plugins/missions.js

```diff
@@ -190,17 +194,137 @@ window.plugin.missions = {
 		}
 	},
 
-	showMissionListDialog: function(missions) {
-		dialog({
-			html: this.renderMissionList(missions),
-			height: 'auto',
-			width: '400px',
-			collapseCallback: this.collapseFix,
-			expandCallback: this.collapseFix,
-		}).dialog('option', 'buttons', {
-			'Create new mission': function() { open('//missions.ingress.com'); },
-			'OK': function() { $(this).dialog('close'); },
-		});
+	showMissionListDialog: function(missions, caption, isPortalList) {
+
+		this.isShowingPortalList = isPortalList;
+
+		// Check whether dialog is already open
+		let openDialog = window.DIALOGS['dialog-missionsList'];
+		if(openDialog)
+		{
+			// Dialog already there, expand if collpased
+			// NOTE: It would be really cool if the IITC core had some function to read and modify collpased state of dialogs
+			// but it hasn't so we need to access internal information here
+			if(plugin.missions.isMissionListCollapsed)
+			{
+				let dia   = $(openDialog).closest('.ui-dialog');
+				let button   = dia.find('.ui-dialog-titlebar-button-collapse');
+				if(button)
+				{
+					$(button).click();
+				}
+			}
+		}
+		else
+		{
+			// If dialog is not open, open it
+			dialog({
+				id: 'missionsList',
+				html: this.renderMissionList(missions),
+				height: 'auto',
+				width: '400px',
+				collapseCallback: this.onCollapseMissionList,
+				expandCallback: this.onExpandMissionList,
+				dragStop: this.resizeMissionList,
+				title: caption,
+				buttons:
+				[
+					{
+						text: "Create new mission",
+						click: function() {
+							open('https://missions.ingress.com/');
+						}
+					},
+					{
+						text: "Ok",
+						click: function() {
+							$(this).dialog('close');
+						}
+					},
+					{
+						text: "Missions in view",
+						css: {float: "left", "margin-right": "2px"},
+						click: function() {
+							plugin.missions.openTopMissions();
+						},
+						create: function()
+						{
+							// Store a link to the button so that we can hide or show it.
+							plugin.missions.fromPortalListToNormalListButton = this;
+						}
+					}
+				]
+			});
+
+			this.isMissionListCollapsed = false;
+		};
+
+		// Dialog will be open now
+		openDialog = window.DIALOGS['dialog-missionsList'];
+
+		// Set content and title
+		$(openDialog).html(this.renderMissionList(missions));
+		$(openDialog).dialog({title: caption});
+
+		// When showing list for one portal, show button to switch back to general list, otherwise hide it
+		if(isPortalList)
+		{
+			$(plugin.missions.fromPortalListToNormalListButton).show();
+		}
+		else
+		{
+			$(plugin.missions.fromPortalListToNormalListButton).hide();
+		}
+
+		this.resizeMissionList();
+	},
```

Comment:
Use 'one true brace-style' https://eslint.org/docs/rules/brace-style#1tbs

Reply:
I made another commit to fix all the style guide issues